lib/filter-rules-loading.js

```diff
@@ -0,0 +1,103 @@
+const path = require('path');
+const yaml = require('js-yaml');
+const fs = require('fs');
+
+const logger = require('./log');
+const { Console } = require('console');
+
+const SUPPORTED_IAC_EXTENSIONS = ['tf', 'yaml', 'yml','tpl','json'];
+function nestedCopy(array) {
+  return JSON.parse(JSON.stringify(array));
+}
+
+function injectRulesAtRuntime(filters) {
+  if (process.env.ACCEPT_IAC) {
+    logger.info(
+      { accept: process.env.ACCEPT_IAC },
+      'Injecting Accept rules for IAC extensions - Possible values tf, yaml, yml, json, tpl',
+    );
+    const extensions = process.env.ACCEPT_IAC.replace(/\s/g, '')
+      .split(',')
+      .filter((extension) => SUPPORTED_IAC_EXTENSIONS.includes(extension));
+    if (extensions.length <= 0) {
+      logger.error(
+        { accept: extensions },
+        '[MISCONFIGURATION] None of the requested ACCEPT IAC file extensions is compatible',
+      );
+    } else {
+      let template = nestedCopy(
+        filters.private.filter(
+          (entry) =>
+            entry.method == 'GET' && entry.path.includes('requirements'),
+        ),
+      );
+      for (let i = 0; i < template.length; i++) {
+        template[i].path = template[i].path
+          .replace('/requirements', '')
+          .replace('%2Frequirements', '');
+        template[i]['//'] = template[i]['//'].replace(
+          'determine the full dependency tree',
+          'scan IAC files',
+        );
+      }
+      for (let i = 0; i < extensions.length; i++) {
+        const extensionTemplate = nestedCopy(template);
+        extensionTemplate[0].path = extensionTemplate[0].path.replace(
+          'txt',
+          extensions[i],
+        );
+        extensionTemplate[1].path = extensionTemplate[1].path.replace(
+          'txt',
+          extensions[i],
+        );
+        filters.private.push(...extensionTemplate);
+      }
+    }
+  }
+  if (process.env.ACCEPT_CODE) {
+    logger.info(
+      { accept: process.env.ACCEPT_CODE },
+      'Injecting Accept rules for Code',
+    );
+    let templateGET = nestedCopy(
+      filters.private.filter((entry) => entry.method == 'GET')[0],
```

Comment:
related to my previous comment, I think that the root cause is this logic as it takes the first `GET` pattern from the rule set and applies it. That pattern is likely taken from request to SCM's API but, in our case, we want to make request to the SCMs host.

Here are the rules, that we verified and that work: https://github.com/snyk/broker/pull/423/files

Reply:
yes I need to distinguish between the different endpoint in used indeed. Will update, thanks for catching that.

Reply:
should be fixed. Going to review all the snapshots to verify their validity before going further